File: SOLID/SingleResponsibilityPrinciple/TypeScript/NewsView.js
"use strict";
exports.__esModule = true;
var NewsView = /** @class */ (function () {
    function NewsView(news) {
        this.news = news;
    }
    NewsView.prototype.html = function () {
        var _a = this.news, title = _a.title, text = _a.text;
        return "\n        <div>\n            <h1>" + title + "</h1>\n            <p>" + text + "</p>\n        </div>\n        ";
    };
    NewsView.prototype.json = function () {
        var _a = this.news, title = _a.title, text = _a.text, changed = _a.changed;
        return JSON.stringify({ title: title, text: text, changed: changed }, null, 2);
    };
    NewsView.prototype.xml = function () {
        var _a = this.news, title = _a.title, text = _a.text, changed = _a.changed;
        return "<news>" + text + "</news>";
    };
    return NewsView;
}());
exports.NewsView = NewsView;

File: SOLID/SingleResponsibilityPrinciple/python/NewsView.py
import json
from SOLID.SingleResponsibilityPrinciple.python.News import News

class NewsView:
    news: News

    def __init__(self, news):
        self.news = news

    def html(self):
        return f"<div><h1>{self.news.title}</h1><p>{self.news.text}</p></div>"

    def json(self):
        obj = {
            'title': self.news.title,
            'text': self.news.text,
            'changed': self.news.changed
        }
        return json.dumps(obj)